src/box/index.cc

```diff
@@ -457,23 +458,58 @@ box_index_iterator(uint32_t space_id, uint32_t index_id, int type,
 	uint32_t part_count = mp_decode_array(&key);
 	if (key_validate(index->def, itype, key, part_count))
 		return NULL;
+	const char *pos, *pos_end;
+	char *buf = NULL;
+	if (packed_pos != NULL && packed_pos != packed_pos_end) {
+		uint32_t buf_size =
+			iterator_position_unpack_bufsize(packed_pos,
+							 packed_pos_end);
+		buf = (char *)xmalloc(buf_size);
+		if (iterator_position_unpack(packed_pos, packed_pos_end,
+					     buf, buf_size,
+					     &pos, &pos_end) != 0) {
+			free(buf);
+			return NULL;
+		}
+		uint32_t pos_part_count = mp_decode_array(&pos);
+		if (exact_key_validate_nullable(index->def->cmp_def, pos,
+						pos_part_count) != 0) {
+			free(buf);
+			return NULL;
+		}
+	} else {
+		pos = NULL;
+		pos_end = NULL;
+	}
 	box_run_on_select(space, index, itype, key_array);
 	struct txn *txn;
 	struct txn_ro_savepoint svp;
-	if (txn_begin_ro_stmt(space, &txn, &svp) != 0)
+	if (txn_begin_ro_stmt(space, &txn, &svp) != 0) {
+		free(buf);
 		return NULL;
-	struct iterator *it = index_create_iterator(index, itype,
-						    key, part_count);
+	}
+	struct iterator *it = index_create_iterator_after(index, itype, key,
+							  part_count, pos);
 	if (it == NULL) {
+		free(buf);
```

Comment:
Nit: Please use `make_scoped_guard` instead of inserting `free(buf)` before each return statement.

Reply:
Done